### DISCUSSION TITLE
Form not revalidating onChange when using Field Arrays and yup.test

### DISCUSSION THREAD

**Discussion by Motoxpro** [react-hook-form/react-hook-form#1745] [Q&A] [answered]
I'm not sure if this is a Yup issue or if I am doing something wrong with RHF. 

I am trying to NOT allow duplicates when using FieldArrays. 

To recreate the error:
1. Append a new field.
2. Change the first name to Bill in both inputs
3. Hit Submit
4. Change the first name to something else and the error persist when it should be cleared.

If you re-submit the form again after the change, the error will be cleared. It just doesn't seem to work onChange

https://codesandbox.io/s/react-hook-form-usefieldarray-l07j9?file=/src/index.js

**Comment by Motoxpro** [comment, marked as answer]
Realized I can clear the error by clearing the top level `test`, but there still seems to be something missing. It doesn't re-validate onchange after that.

**Reply by bluebill1049** [reply]
React hook form validate input at a time to give you perf edge instead validate the entire form, what you try to validate is a form level validation. you will have to trigger validation for the entire form on input change.

try this with onChange `triggerValidation()`

**Reply by Motoxpro** [reply]
Thanks @bluebill1049 ! It didn't seem to work when validating the whole form with `triggerValidation()` but it did when validating just the field arrays for some reason `triggerValidation('test')`

Thanks again! I was happy to be able to donate some $ to the project this month!

**Reply by Motoxpro** [reply]
@bluebill1049 One more thing :(

If there is an error while using test, you get the `data.slice is not a function or its return value is not iterable` error.

To see what I'm talking about:
1. Add another field
2. Change both first names to `Bill`
3. Hit submit
4. Delete one of the fields while the error is still there.

**Reply by bluebill1049** [reply]
oh sorry, completely miss this one... I couldn't reproduce the problem with the steps above perhaps i am missing something. also i see your CSB is on 5.5 maybe try 5.72 see if that resolves the problem.

**Reply by Motoxpro** [reply]
No worries, Bill! Looks like it is fixed in the latest version (6.0.1). Thank you!

**Reply by bluebill1049** [reply]
awesome! thanks 🙏
